Interface Class Specifications/Display/InterfaceSpec-Display.md

```diff
@@ -0,0 +1,86 @@
+# Interface Class Specification: Display
+> Status: Draft Proposal
+
+## Device class summary
+The display interface is responsible for filling the role previously filled by static display config json in osvr_server_config.json. Where previously the user would be responsible for selecting the correct display configuration, the display interface allows hardware plugins to specify these parameters automatically at runtime.
+
+### Examples
+- None
+
+
+### Relation to other classes
+**Factoring**: The display interface represents a collection of individual parameters for a given display, including parameters that don't fit neatly into an existing interface. While it might be possible (and in some cases appropriate) to expose certain display parameters individually as an analog or button type interface, other types of parameters, such as the distortion mesh or the logical display topology (viewer/eye/surface mapping to physical displays and viewports), cannot.
+
+**From other instances of the same interface**: It isn't likely that analysis plugins will consume a display interface and filter or transform any of its parameters, although it remains possible.
+
+**From other classes**: Although not expected to be common, it may be possible to derive a display interface out of a set of other interfaces representing individual parameters.
+
+**To other classes**: If some display parameters are dynamic, and the hardware in question can read the value of the parameter at runtime as it changes, then it may be useful to expose that parameter separately as its own interface. For example, a display's center of projection might be exposed as an `OSVR_Vec2` interface.
+
+## Messages
+In general, to work smoothly with the rest of the OSVR system, distance units are meters and velocity units are meters/second.
+
+### Display
+#### Data
+- type (enum in fixed|hmd)
+- vendor (string)
+- model: (string)
+- product URI (uri-string RFC 3986)
+- driver URI (uri-string RFC 3986)
+- notes: string
+- number of DisplayInputs (whole number)
+- number of logical DisplaySurfaces (whole number)
+- k1[3] rgb radial distortion supported (boolean, hmd only?) and priority (whole number)
+- RGB radial distortion supported (boolean, hmd only?, supercedes k1[3] in most cases) and priority (whole number)
+- mono point mesh distortion supported (boolean, hmd only?) and priority (whole number)
+- RGB mesh distortion supported (boolean, hmd only?) and priority (whole number)
+
+#### Rationale
+The Display message is the root object of the Display interface. It references both the physical DisplayInputs and logical DisplaySurfaces, which define the mapping between them.
+
+### DisplayInput
+> one per number of display inputs in Display message.
+
+#### Data
+- resolution (width, height in whole numbers)
```

Comment:
Something we can't detect, though, is the scan-out origin in the supported display resolutions.

For instance, the first pixel scanned out in the HDK 1.x is in the upper left when it's in 1920×1080 and the upper right when it's in 1080×1920. In the HDK 2, the first pixel is in the lower right.

The combination of the position of this pixel and the display mode (horizontal side-by-side, vertical side-by-side, or full-screen) [both provided by descriptor] plus the current resolution and rotation [provided by display detection API] can be used to determine the orientation of the display and the eye viewport parameters.